mlflow/utils/autologging_utils/safety.py

```diff
@@ -833,49 +963,73 @@ def _validate(autologging_call_input, user_call_input=None):
-                )
-            )
+            _assert_same_key_set(autologging_call_input, user_call_input)
             for key in autologging_call_input.keys():
                 _validate(autologging_call_input[key], user_call_input.get(key, None))
         else:
-            assert (
-                autologging_call_input is user_call_input
-                or autologging_call_input == user_call_input
-            ), (
-                "Input to original function does not match expected input."
-                " Original: '{}'. Expected: '{}'".format(autologging_call_input, user_call_input)
-            )
+            _assert_equivalence(autologging_call_input, user_call_input)
+
+    # Similar validation logic found in _validate, unraveling the list of arguments to exclude
+    # checks for any validation exempt positional arguments.
+    if user_call_args is None and autologging_call_args is not None:
+        _validate_new_input(autologging_call_args)
+    else:
+        _assert_same_type(autologging_call_args, user_call_args)
+        if autologging_call_args is None:
+            _assert_equivalence(autologging_call_args, user_call_args)
+        else:
+            _assert_same_input_length(autologging_call_args, user_call_args)
+            for index, autologging_call_arg, user_call_arg in itertools.zip_longest(
+                range(len(user_call_args)), autologging_call_args, user_call_args
+            ):
+                if not _is_arg_exempt_from_validation(
+                    autologging_integration,
+                    function_name,
+                    user_call_arg,
+                    positional_argument_index=index,
+                ):
+                    _validate(autologging_call_arg, user_call_arg)
 
-    _validate(autologging_call_args, user_call_args)
-    _validate(autologging_call_kwargs, user_call_kwargs)
+    # Similar validation logic found in _validate, unraveling the dictionary of arguments to exclude
+    # checks for any validation exempt keyword arguments.
+    if user_call_kwargs is None and autologging_call_kwargs is not None:
+        _validate_new_input(autologging_call_kwargs)
+    else:
+        _assert_same_type(autologging_call_kwargs, user_call_kwargs)
+        if autologging_call_kwargs is None:
+            _assert_equivalence(autologging_call_args, user_call_args)
+        else:
+            _assert_same_key_set(autologging_call_kwargs, user_call_kwargs)
+            for key in autologging_call_kwargs.keys():
+                if not _is_arg_exempt_from_validation(
+                    autologging_integration,
+                    function_name,
+                    user_call_kwargs.get(key, None),
+                    keyword_argument_name=key,
+                ):
+                    _validate(
+                        autologging_call_kwargs[key],
+                        user_call_kwargs.get(key, None),
+                    )
```

Comment:
The set of keyword arguments of a function call is always a dictionary. For simplicity, can we rewrite this logic with that assumption?